src/pages/open-api-docs-preview/index.tsx

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { GetStaticPropsResult } from 'next'
+import OpenApiDocsPreviewView from 'views/open-api-docs-preview'
+
+const IS_PRODUCTION = process.env.NODE_ENV === 'production'
+
+/**
+ * We don't actually need static props for this page,
+ * we use `getStaticProps` here to prevent the page from being rendered
+ * in production.
+ */
+export async function getStaticProps(): Promise<
+	GetStaticPropsResult<Record<string, never>>
+> {
+	/**
+	 * In production, return a 404 not found for this page.
+	 * In other environments (local, preview, and staging), show the page.
+	 */
+	return IS_PRODUCTION ? { notFound: true } : { props: {} }
```

Comment:
I haven't actually tried this before, but in theory this should mean that even after merging this PR to `main`, <https://developer.hashicorp.com/open-api-docs-preview> will `404`, and we'll only see this preview tool in local dev, preview deploys, and perhaps most importantly our staging deploy at <https://developer.hashicorp.services/open-api-docs-preview> 👀